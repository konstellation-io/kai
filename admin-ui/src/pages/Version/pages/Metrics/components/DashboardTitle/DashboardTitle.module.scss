@import '../../../../../../styles/mixins';
@import '../../../../../../styles/colors';

.container {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  height: 61px;
  padding: 0 10px;
}

.title {
  @include font-subheader;
  color: font-color(light);
}
