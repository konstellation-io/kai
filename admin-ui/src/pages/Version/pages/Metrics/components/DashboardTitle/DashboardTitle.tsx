import React from 'react';

import Calendar from '../../../../../../components/Form/Calendar/Calendar';

import styles from './DashboardTitle.module.scss';
import { Moment } from 'moment';

type Props = {
  runtimeName?: string;
  versionName?: string;
  value: Function;
  onChange: Function;
  submit: Function;
};
function DashboardTitle({
  runtimeName,
  versionName,
  value,
  onChange,
  submit
}: Props) {
  return (
    <div className={styles.container}>
      <div className={styles.title}>
        {`Metrics for ${runtimeName} - ${versionName}`}
      </div>
      <div className={styles.dateFilter}>
        <Calendar
          label="filter by dates"
          onChangeFromDateInput={(date: Moment) => onChange('startDate', date)}
          onChangeToDateInput={(date: Moment) => onChange('endDate', date)}
          formFromDate={value('startDate')}
          formToDate={value('endDate')}
          submit={submit}
          addTimeControls
          hideError
        />
      </div>
    </div>
  );
}

export default DashboardTitle;
