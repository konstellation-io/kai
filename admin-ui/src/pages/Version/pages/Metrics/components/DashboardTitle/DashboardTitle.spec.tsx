import React from 'react';
import { shallow } from 'enzyme';
import DashboardTitle from './DashboardTitle';

jest.mock('react-router', () => ({
  useParams: jest.fn(() => ({ runtimeId: '', versionId: '' }))
}));

describe('DashboardTitle', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <DashboardTitle
        runtimeName="runtimeName"
        versionName="versionName"
        onChange={jest.fn()}
        value={jest.fn()}
      />
    );
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
