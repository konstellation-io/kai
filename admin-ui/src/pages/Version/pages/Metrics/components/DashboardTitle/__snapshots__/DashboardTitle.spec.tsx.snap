// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardTitle matches snapshot 1`] = `
<div
  className="container"
>
  <div
    className="title"
  >
    Metrics for runtimeName - versionName
  </div>
  <div
    className="dateFilter"
  >
    <Calendar
      addTimeControls={true}
      hideError={true}
      label="filter by dates"
      onChangeFromDateInput={[Function]}
      onChangeToDateInput={[Function]}
    />
  </div>
</div>
`;
